tests/durability/test_disk_failure_recovery.py

```diff
@@ -520,3 +521,460 @@ def test_sns_repair_fail_disk_diff_cvg_equal_to_k(self):
 
         LOGGER.info("COMPLETED: Test SNS repair works fine with failed disks "
                     "are equal to K(parity units) and from different cvg")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.data_durability
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-36396")
+    def test_near_full_sns_repair_fail_disk_diff_cvg_less_than_k(self):
+        """
+        Validate SNS repair works fine on near full system with failed
+        disks are less than K(parity units)
+        """
+        LOGGER.info("STARTED: Validate SNS repair works fine on near full system with failed disks "
+                    "are less than K(parity units)")
+        near_full_percent = 60
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.s3_clean = users
+        workload_info = None
+        s3userinfo = list(users.values())[0]
+
+        LOGGER.info("Step 1: Perform Write operations till overall disk space is filled %s",
+                    near_full_percent)
+        resp = self.dsk_rec_obj.get_user_data_space_in_bytes(self.node_master_list[0],
+                                                             near_full_percent)
+        assert_utils.assert_true(resp[0], resp[1])
+
+        if not resp[1]:
+            LOGGER.info("Current Memory usage is already more than expected memory usage,"
+                        " skipping write operation")
+        else:
+            workload_info = self.dsk_rec_obj.perform_near_full_sys_writes(s3userinfo=s3userinfo,
+                                                                          user_data_writes=resp[1],
+                                                                          bucket_prefix=
+                                                                          self.test_prefix[-1])
```

Comment:
This will also delete all buckets which were written. Is that expected?

Reply:
Made skip_cleanup=True in perform_near_full_sys_writes